apps/address-book/app/store/entry.js

```diff
@@ -51,21 +51,15 @@ export const onEntryUpdated = async ({ entries = [] }, { addr }) => {
 /*    AddressBook helper functions    */
 /// /////////////////////////////////////
 
-const loadEntryData = (addr) => {
-  return new Promise((resolve, reject) => {
+const loadEntryData = addr => {
+  return new Promise(resolve => {
     app.call('getEntry', addr).subscribe(async cid => {
-      if (!cid) resolve() // entry probably removed in a future block
```

Comment:
Interesting that JS will continue to evaluate the function body after `resolve` is called. Good catch.